utils/src/callWithTelemetryAndErrorHandling.ts

```diff
@@ -237,6 +237,13 @@ function handleTelemetry(context: types.IActionContext, callbackId: string, star
                     }
                 }
             }
+            if (context.telemetry.properties.subscriptionId) {
+                // set the lastReported subscription id to the current one if it exists
+                ext._internalReporter.lastReportedAzureSubscriptionId = context.telemetry.properties.subscriptionId;
+            } else {
+                // use the last reported subscription id if no subscription id was set on the telemetry properties
+                context.telemetry.properties.subscriptionId = ext._internalReporter.lastReportedAzureSubscriptionId;
+            }
```

Comment:
This seems like something that is better solved at the time of Kusto querying, rather than this. Failing that, I'd try to create a telemetry handler (https://github.com/microsoft/vscode-azuretools/blob/39c8500374abeb2c82a19d81c968ecaeb167bf38/utils/src/callWithTelemetryAndErrorHandling.ts#L97-L99) to do this if possible.